js/modules/SearchTokenizer/SearchInput.js

```diff
@@ -55,20 +59,64 @@ export default class SearchInput {
             backspace_action: 'edit',
             tokenizer_options: {},
             filter_on_type: true,
+            input_options: {
+                classes: [],
+                attributes: {},
+                data: {}
+            }
         }, options || {});
         this.tokenizer = new SearchTokenizer(options.allowed_tags || {}, options.drop_unallowed_tags || false, options.tokenizer_options);
 
         this.displayed_input = $(`
          <div class="form-control search-input d-flex" tabindex="0"></div>
       `).insertBefore(input);
         this.displayed_input.append(`<span class="search-input-tag-input flex-grow-1" contenteditable="true"></span>`);
+        this.applyInputOptions();
+
         this.original_input.hide();
 
         this.last_result = null;
 
         this.registerListeners();
     }
 
+    applyInputOptions() {
+        let new_attrs = {};
+
+        if (typeof this.options.input_options.attributes === 'object') {
+            new_attrs = this.options.input_options.attributes;
+        } else if (this.options.input_options.attributes === 'copy') {
+            const original_attr = this.original_input.attr();
+            // Get only non-data attributes
+            new_attrs = Object.keys(original_attr).filter(key => !key.startsWith('data-')).reduce((obj, key) => {
+                obj[key] = original_attr[key];
+                return obj;
+            }, {});
+        }
+
+        let new_data = {};
+        if (typeof this.options.input_options.data === 'object') {
+            new_data = this.options.input_options.data;
+        } else if (this.options.input_options.data === 'copy') {
+            new_data = this.original_input.data();
+        }
+        // Add data attributes. We don't use $.data() because having the DOM attribute may be needed and using $.data doesn't add them.
+        Object.assign(Object.keys(new_data).reduce((obj, key) => {
+            obj['data-' + key] = new_data[key];
+            return obj;
+        }, new_attrs));
+
+        // Apply attributes including data attributes
+        this.displayed_input.attr(new_attrs);
+
+        // Apply classes
+        if (Array.isArray(this.options.input_options.classes)) {
+            this.displayed_input.addClass(this.options.input_options.classes.join(' '));
+        } else if (this.options.input_options.classes === 'copy') {
+            this.displayed_input.addClass(this.original_input.attr('class').join(' '));
```

Comment:
`$.attr('class')` will return a string.
```suggestion
            this.displayed_input.addClass(this.original_input.attr('class'));
```